rtl/eclrtl/eclrtl.cpp

```diff
@@ -4892,6 +4894,50 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         }
     }
 
+    bool nextMatch()
+    {
+        if (!matched)
+            return false;
+
+        const char *start = subs[0].second==subs[0].first?subs[0].second+1: subs[0].second;
+        if (start >= sample_end)
+            return (matched = false);
+
+        matched = boost::regex_search(start, subs, *regEx);
+
+        return matched;
+    }
+
+    void getMatchSet(bool  & __isAllResult, size32_t & __resultBytes, void * & __result)
+    {
+        rtlRowBuilder out;
+        byte *outData;
+        size32_t outBytes = 0;
+
+        boost::regex_iterator<char *> cur(sample, sample_end, *regEx);
+        boost::regex_iterator<char *> end;
+
+        for (; cur != end; ++cur)
+        {
+            const boost::match_results<char *> &match = *cur;
+
+            if (match[0].first==sample_end) break;
+            rtlDataAttr str;
+            size32_t lenBytes = match[0].second - match[0].first;
+
+            out.ensureAvailable(outBytes + lenBytes + sizeof(size32_t));
```

Comment:
Probably better to do in two passes.  The first pass calculates the size of the result.  Then allocate the return structure and fill it in in a second pass.